Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/Cql2ElmVisitor.java

```diff
@@ -2746,7 +2746,9 @@ public Object visitAggregateExpressionTerm(cqlParser.AggregateExpressionTermCont
     @Override
     public Object visitSetAggregateExpressionTerm(cqlParser.SetAggregateExpressionTermContext ctx) {
         Expression source = parseExpression(ctx.expression(0));
-        Expression per = null;
+
+        // If `per` is not set, it will remain `null as System.Quantity`.
+        Expression per = libraryBuilder.buildNull(libraryBuilder.resolveTypeName("System", "Quantity"));
```

Comment:
question: Seems like this actually move to line 2757? Ideally, we're not setting `per` more than once, because that implies we're doing (potentially) unnecessary work?

Reply:
Hi @JPercival I agree this adds unnecessary work when `per` is later reassigned (reassignment itself is also bad..). I chose to do it this way because otherwise we'd repeat `libraryBuilder.buildNull(libraryBuilder.resolveTypeName("System", "Quantity"));` multiple times.